src/main/java/kitchenpos/menus/tobe/menu/application/MenuService.java

```diff
@@ -0,0 +1,74 @@
+package kitchenpos.menus.tobe.menu.application;
+
+import kitchenpos.menus.tobe.menu.application.dto.MenuCreationRequestDto;
+import kitchenpos.menus.tobe.menu.application.dto.MenuCreationResponseDto;
+import kitchenpos.menus.tobe.menu.application.dto.ProductQuantityDto;
+import kitchenpos.menus.tobe.menu.application.exception.MenuGroupNotExistsException;
+import kitchenpos.menus.tobe.menu.domain.*;
+import kitchenpos.menus.tobe.menuGroup.application.MenuGroupService;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+public class MenuService {
+
+    private final MenuRepository menuRepository;
+    private final Products products;
+    private final MenuGroupService menuGroupService;
+
+    public MenuService(final MenuRepository menuRepository, final Products products, final MenuGroupService menuGroupService) {
+        this.menuRepository = menuRepository;
+        this.products = products;
+        this.menuGroupService = menuGroupService;
+    }
+
+    public MenuCreationResponseDto create(final MenuCreationRequestDto menuCreationRequestDto) {
+
+        try {
+            menuGroupService.findById(menuCreationRequestDto.getMenuGroupId());
+        } catch (IllegalArgumentException e) {
+            throw new MenuGroupNotExistsException();
+        }
+
+        // 제품 미지정 시 에러
+        if (menuCreationRequestDto.getProductQuantityDtos() == null || menuCreationRequestDto.getProductQuantityDtos().isEmpty()) {
+            throw new IllegalArgumentException("메뉴 내 제품을 1개 이상 지정해야합니다.");
+        }
+
+        // 제품 중복 시 에러
+        if (menuCreationRequestDto.getProductQuantityDtos().size() != menuCreationRequestDto.getProductQuantityDtos()
+                .stream()
+                .map(ProductQuantityDto::getProductId)
+                .distinct()
+                .count()) {
+            throw new IllegalArgumentException("메뉴 내 제품은 중복될 수 없습니다.");
+        }
+
+        // 제품 Id 리스트 생성
+        final List<Long> productIds = menuCreationRequestDto.getProductQuantityDtos()
+                .stream()
+                .map(p -> p.getProductId())
+                .collect(Collectors.toList());
+
+        // 제품 수량 리스트 생성
+        final List<Long> quantities = menuCreationRequestDto.getProductQuantityDtos()
+                .stream()
+                .map(p -> p.getQuantity())
+                .collect(Collectors.toList());
+
+        // MenuProduct 정보 가져오기 (Anti-Corruption Layer)
+        final List<MenuProduct> menuProducts = products.getMenuProductsByProductIdsAndQuantities(productIds, quantities);
```

Comment:
지금은 productIds와 quantities의 길이가 맞는지 체크하는 부분도 번거롭고, 따로따로 넘기는 부분도 조금 번거로워서 코드가 조금 혼란스러워 지는 것 같아요 그냥 구현되어 있는 `productQuantityDtos` 리스트를 바로 넘겨버린 뒤 `List<MenuProduct>`로 만들어주는게 조금 더 나을 것 같네요.

```suggestion
        // MenuProduct 정보 가져오기 (Anti-Corruption Layer)
        final List<MenuProduct> menuProducts = products.getMenuProductsByProductIdsAndQuantities(menuCreationRequestDto.getProductQuantityDtos());
```

위에 따로 코멘트를 남기긴 했는데.. productIds만 받아서 `List<Product>`를 내려준 뒤 MenuService 안에서 작업을 하는게 좋을지, `List<MenuProduct>`를 내리는게 좋을지 고민이 되네요 ㅠㅠ 이 부분은 더 낫다고 생각되어지는 구현을 하시는게 좋을 듯 합니다.

Reply:
Products Anti-Corruption Layer에서는 다른 context의 엔티티를 그대로 사용하는 것이 덜 헷갈릴 것 같아서 결국.. List<Product>를 내려주는 방식으로 구현했습니다..!

Reply:
추가로 이 부분은 저도 조금 고민이 되어서, 다른 리뷰어분들에게도 의견을 물어보았어요

```
Product는 다른 CONTEXT의 AGGREGATE이기 때문에 Product 또한 반환되는 것은 옳지 않아 보여요. (물론 메뉴 CONTEXT에서 도출된 Product라면 가능합니다.)

그러면 문제되는 부분이 quantity인데, 이 경우 `id`와 name을 가진 VALUE OBJECT와 그것을 포함하는 VALUE OBJECT를 만들 것 같아요!
```

위와 같은 답변을 받았는데, 참고하시면 좋을 것 같아요!

Reply:
해당 내용을 반영해 주셔도 좋고, 아니면 이번엔 참고만 하신 다음에 다음에 수정하셔도 될 것 같아요.

Reply:
> 추가로 이 부분은 저도 조금 고민이 되어서, 다른 리뷰어분들에게도 의견을 물어보았어요
> 
> ```
> Product는 다른 CONTEXT의 AGGREGATE이기 때문에 Product 또한 반환되는 것은 옳지 않아 보여요. (물론 메뉴 CONTEXT에서 도출된 Product라면 가능합니다.)
> 
> 그러면 문제되는 부분이 quantity인데, 이 경우 `id`와 name을 가진 VALUE OBJECT와 그것을 포함하는 VALUE OBJECT를 만들 것 같아요!
> ```
> 
> 위와 같은 답변을 받았는데, 참고하시면 좋을 것 같아요!

MenuService에서는 Anti-Corruption Layer에서 Product의 'id', 'price'만 받아오면 되는 상황인데, 해당 값들을 포함한 Value Object를 말씀하시는건가요?!

Reply:
앗, 윗 부분만 읽고 아랫 부분은 제대로 읽어보지 않았었네요 (위에서 Product가 반환되는것이 옳지 않은것 같다고 의견을 주신 순간에 바로 넘어가버렸습니다 ㅠㅠ)

저는 Product가 반환되지 않아야 하므로 `MenuProduct`를 반환하는 방향으로 하고, `id`와 `quantity`를 가진 ValueObject를 전달해주는것으로 저는 이해했어요

 Anti-Corruption Layer에서 Product와 관련된 값들을 포함한 Value Object를 생성하면 결국 다시 MenuService에서 받아서 변환해주는 작업이 필요하게 될 것 같아서 저는 위에 말했던 방향이 좋다고 생각합니다.